roofit/roofit/src/RooBernstein.cxx

```diff
@@ -228,7 +228,7 @@ RooSpan<double> RooBernstein::evaluateBatch(std::size_t begin, std::size_t batch
 
 Int_t RooBernstein::getAnalyticalIntegral(RooArgSet& allVars, RooArgSet& analVars, const char* rangeName) const
 {
-  if (rangeName && strlen(rangeName))   return 0;
+  if (rangeName && strlen(rangeName))   return 1;
 
   if (matchArgs(allVars, analVars, _x)) return 1;
```

Comment:
I'm not sure this does what's intended. Let me try to spell out what I think this code does:
- If the user is giving a range name for the integral, an analytical integral is used, irrespective of what the integrand is.
- If the integral is over the default range (no name given), an analytical integral is used if `x` is the integrand.
- Otherwise, a numerical integral is used.

I suspect that range checking is not necessary here. We should just check if we are integrating over `x` or not.